packages/block-editor/src/components/inner-blocks/use-block-context.js

```diff
@@ -25,7 +25,7 @@ export default function useBlockContext( clientId ) {
 	return useSelect(
 		( select ) => {
 			const block = select( blockEditorStore ).getBlock( clientId );
-			const blockType = getBlockType( block.name );
+			const blockType = getBlockType( block?.name );
```

Comment:
The `getBlockType` will return `undefined` when there's no block name, and we'll bail out early below. Let me know if I should add a more explicit bailout for missing a block object.

Reply:
Personally I would prefer explicit bailouts. If `! block`, return `undefined`. If `! blockType`, return `undefined`. It's better style than all the `?.`'s and `??`'s.

Reply:
Would it be preferable to get the block type from data store, i.e., `select( blocksStore ).getBlockType()`? In theory it takes into account a custom data registry, instead of using a global singleton.

Reply:
Updated code to use early returns and `getBlockType` from store 👍